### DISCUSSION TITLE
SlidingWindowInferer vs PatchInferer

### DISCUSSION THREAD

**Discussion by solarflarefx** [Project-MONAI/MONAI#7251] [Q&A]
Hi all,

I wanted to better understand when to use SlidingWindowInferer vs PatchInferer.  

- I looked at the example tutorial code and I only saw PatchInferer used for a 2D example, but from the documentation it seems to be it would also work in 3D.
- Another thought that came to mind was that perhaps the SlidingWindowInferer allowed for overlaps while PatchInferer does not, but the example provided for PatchInferer also uses overlaps.

**Comment by KumoLiu** [comment]
Hi @solarflarefx, thanks for your interest here!

`PatchInferer` is introduced by MONAI 1.2 which is a more general concept than `SlidingWindowInferer`. It employs and coordinates different pieces to provide flexibility and extensibility for patch-wise inference and merging their results to create a merged output.
For more reasons for designing `PatchInferer`, you could refer to this [issue](https://github.com/Project-MONAI/MONAI/issues/5837).
The reason why `SlidingWindowInferer` is still around is probably because it's a very popular class at MONAI, so we don't want to move it easily.
